lua/evergarden/init.lua

```diff
@@ -38,43 +38,44 @@ end
 ---@param group string
 ---@param colors evergarden.types.colorspec
 local function set_hi(group, colors)
-  if type(colors) ~= 'table' or vim.tbl_isempty(colors) then
-    return
-  end
-
-  colors.fg = colors.fg or colors[1] or 'none'
-  colors.bg = colors.bg or colors[2] or 'none'
-
-  ---@type vim.api.keyset.highlight
-  local color = vim.deepcopy(colors)
-
-  color.fg = type(colors.fg) == 'table' and colors.fg[1] or colors.fg
-  color.bg = type(colors.bg) == 'table' and colors.bg[1] or colors.bg
-  color.ctermfg = type(colors.fg) == 'table' and colors.fg[2] or 'none'
-  color.ctermbg = type(colors.bg) == 'table' and colors.bg[2] or 'none'
-  color[1] = nil
-  color[2] = nil
-  color.name = nil
-
-  vim.api.nvim_set_hl(0, group, color)
+    if not vim.tbl_isempty(colors) then
+        ---@type vim.api.keyset.highlight
+        local color = colors
+
+        -- priority colors[str] > colors[num][num] > colors[num] )
+        -- e.g. colors.fg > colors[1][1] > colors[1] 
+        if colors[1] then
+            if vim.tbl_islist(colors[1]) then
+                color.fg = colors.fg or colors[1][1] or 'NONE'
+                color.ctermfg = colors.ctermfg or colors[1][2] or 'NONE'
+            else
+                color.fg = colors.fg or colors[1]
+            end
+        end
+
+        if colors[2] then
+            if vim.tbl_islist(colors[2]) then
+                color.bg = colors.bg or colors[2][1] or 'NONE'
+                color.ctermbg = colors.ctermbg and colors[2][2] or 'NONE'
+            else
+                color.bg = colors.bg or colors[2]
+            end
+        end
```

Comment:
this won't work. i just made a few tests based on your examples and this will result in an invalid value:
```lua
{ { '#caffee' }, fg = { '#caffee' } }
```
here set_hl will use `{ '#caffee' }` instead of the expected `'#caffee'`

Reply:
these tests do all pass for the previous implementation of `set_hl`, so im struggling to understand what you are trying to fix.